File: Task 1/Java/app/src/main/java/dev/ogabek/java/MainActivity.java
package dev.ogabek.java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

    }

    void initViews() {
        btn = findViewById(R.id.button);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                shareInfoToInformationActivity();
            }
        });
    }

    void shareInfoToInformationActivity() {
        Intent intent = new Intent(this, InformationActivity.class);
        intent.putExtra("name", "Ogabek");
        intent.putExtra("surname", "Matyakubov");
        startActivity(intent);
    }

}

File: Task 1/Kotlin/app/src/main/java/dev/ogabek/kotlin/MainActivity.kt
package dev.ogabek.kotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {

    private lateinit var btn: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

    }

    private fun initViews() {
        btn = findViewById(R.id.button)
        btn.setOnClickListener {
            shareInfoToInformationActivity()
        }
    }

    private fun shareInfoToInformationActivity() {
        val intent = Intent(this, InformationActivity::class.java)
        intent.putExtra("name", "Ogabek")
        intent.putExtra("surname", "Matyakubov")
        startActivity(intent)
    }

}